.semaphore/collect-artifacts-from-vms

```diff
@@ -28,10 +28,32 @@ echo "Collecting artifacts..."
 num_fv_batches=${NUM_FV_BATCHES:-8}
 batches=(ut $(seq 1 ${num_fv_batches}))
 pids=()
-monitor_pids=()
+log_files=()
 for batch in "${batches[@]}"; do
   vm_name="$vm_name_prefix$batch"
-  echo "Collecting artifacts from $vm_name..."
-  VM_NAME=$vm_name $my_dir/on-test-vm ${REPO_NAME}/.semaphore/collect-artifacts || echo "Failed to collect artifacts from ${vm_name}."
-  gcloud --quiet compute scp "--zone=${ZONE}" "ubuntu@${vm_name}:${REPO_NAME}/artifacts" ./ --recurse || echo "Failed to SCP from ${vm_name}"
+  log_file="$artifacts_dir/collect-artifacts-$batch.log"
+  log_files+=( $log_file )
+  echo "Collecting artifacts from $vm_name. Logging to $log_file."
+  (
+    args=""
+    # Only collect binaries once since they're large to copy and should be identical.
+    if [ $batch != 1 ]; then
+      args="--skip-binaries"
+    fi
+    VM_NAME=$vm_name $my_dir/on-test-vm ${REPO_NAME}/.semaphore/collect-artifacts $args || echo "Failed to collect artifacts from ${vm_name}."
+    gcloud --quiet compute scp "--zone=${ZONE}" "ubuntu@${vm_name}:${REPO_NAME}/artifacts" ./ --recurse || echo "Failed to SCP from ${vm_name}"
+  ) > "$log_file" &
+  pid=$!
+  pids+=( $pid )
+done
+
+echo "===== Waiting for diags collection to complete ====="
+for (( i=0; i<num_batches; i++ )); do
+  pid=${pids[$i]}
```

Comment:
Check here that $pid is non-empty?

Reply:
Are you thinking just an assert?  I think the code would have to be bugged to have an empty value here

Reply:
Yes I was thinking an assert.  So maybe not worth it after all.